script.js

```diff
@@ -1,4 +1,183 @@
+    let player = `Player ${i + 1}`;
+    leaderboard.push({ player });
+    leaderboard[i].value = arr[i];
+  }
+  if (mode == "big") {
+    record = leaderboard.sort((a, b) => (a.value < b.value ? 1 : -1));
+  } else {
+    record = leaderboard.sort((a, b) => (a.value > b.value ? 1 : -1));
+  }
+  for (let j = 0; j < record.length; j++) {
+    showRecord +=
+      `${[j + 1]}. ` + record[j].player + " : " + record[j].value + `<br>`;
+  }
+  return showRecord;
+}
+
+// Find the index of duplicated number in game board array
+function findDuplicatesIndex(arr, num) {
+  let duplicatesIndex = [];
+  let idx = arr.indexOf(num);
+  while (idx != -1) {
+    duplicatesIndex.push(idx);
+    idx = arr.indexOf(num, idx + 1);
+  }
+  return duplicatesIndex;
+}
+
+// Check winner and display
+function checkWinner(numOfDice, numOfPlayer, gameMode) {
+  if (playerCounter <= numOfPlayer) {
+    gameRecordArr.push(getDiceCombo(numOfDice, gameMode));
+    for (let i = 0; i < gameRecordArr.length; i++) {
+      let diceNumsStr = "";
+      for (let j = 0; j < numsArr.length; j++) {
+        diceNumsStr += " " + numsArr[j] + " ";
+      }
+      // Display each player's dice number and optimal number combo
+      diceRollMsg = `Player ${playerCounter}, you roll "${diceNumsStr}", your number is ${gameRecordArr[i]}<hr>`;
+    }
+
+    gameBoard = generateLeaderboard(gameRecordArr, gameMode);
+    playerCounter++;
+    submitBtn.innerHTML = `Player ${playerCounter}`;
+
+    if (playerCounter > numOfPlayer) {
+      let maxNum = Math.max(...gameRecordArr);
+      let minNum = Math.min(...gameRecordArr);
+
+      // Find the index of maxNum if have more then two maxNum
+      let dulNum =
+        gameMode == "big"
+          ? Math.max(...gameRecordArr)
+          : Math.min(...gameRecordArr);
+      let duplicatedNumIndexArr = findDuplicatesIndex(gameRecordArr, dulNum);
+      let dulTopPlayers = "";
+
+      // Display duplicated players which have the highest number
+      if (duplicatedNumIndexArr.length >= 2) {
+        for (let i = 0; i < duplicatedNumIndexArr.length; i++) {
+          dulTopPlayers += `Player ${duplicatedNumIndexArr[i] + 1}, `;
+        }
+        toggleBtn();
+        return `It's a tie! ${dulTopPlayers}<br>you have the same number: ${dulNum}. `;
+      }
+
+      // Display winner
+      if (gameMode == "big" && maxNum != minNum) {
+        winner = gameRecordArr.indexOf(maxNum) + 1;
+      } else {
+        winner = gameRecordArr.indexOf(minNum) + 1;
+      }
+
+      winnerMsg = `Congrats! Player ${winner}, you won!!!🎊🎊🎊`;
+      displayMsg = `${diceRollMsg}<br>${winnerMsg}<br><br>`;
+      toggleBtn();
+    } else {
+      displayMsg = diceRollMsg;
+    }
+  }
+  return displayMsg + "<b>🏆Leaderboard🏆</b><br>" + gameBoard;
+}
+
+// Main function block
+let main = function () {
+  const num_Of_Dice = diceInput.value;
+  const num_Of_Player = playerInput.value;
+  const game_Mode = gameModeInput.value;
```

Comment:
rmb variables in javascript use camelCase

theres an alternate convention for constants which uses SCREAM_CASE